### PR TITLE
Feature: Random package

### PR DESCRIPTION
**Pull request by guisithos**
Besides what was mentioned in issue #28, I took the liberty of also adding an option to handle slices with Pick() and Shuffle().

Example of use:

```
// Generate a secure random string (like for an API key)
key, err := rand.StringWithLength(32)
if err != nil {
    return err
}

// Randomly select a winner from a list
participants := []string{"Alice", "Bob", "Charlie"}
winner, err := rand.Pick(participants)
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a package for generating random numbers and strings with various methods for number generation, string creation, element picking, and shuffling.
	- Added customizable options for string length and character sets.

- **Tests**
	- Implemented a comprehensive suite of unit tests to ensure the correctness and reliability of random generation functionalities.

### CODE REVIEW

File: rand/rand.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return StringWithLength(DefaultLength)
+}
+
+// StringWithLength generates a random string of the x length using the default charset
+func StringWithLength(length int) (string, error) {
+    return StringWithCharset(length, DefaultCharset)
+}
+
+// Pick returns a random element from the provided slice
+func Pick[T any](slice []T) (T, error) {
```
Comment (kashifkhan0771) [inline rand/rand.go:60]: Thanks for adding these additional Functions ❤️

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]
I just took a quick look. Overall looks ok. I resolved/committed some suggestions from coderabbitai. Please check other comments from bot and If they can be resolved.

I'll do a more detail review later on. Thank you for your contribution!

File: rand/rand.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+func Number() (int64, error) {
+	n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		return 0, fmt.Errorf("failed to generate random number: %w", err)
+	}
+	return n.Int64(), nil
+}
+
+// NumberInRange generates a random number between min and max
+func NumberInRange(min, max int64) (int64, error) {
```
Comment (shahzadhaider1) [inline rand/rand.go:27]: good to add an early case: If min == max

File: rand/rand.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	if min > max {
+		return 0, fmt.Errorf("min (%d) cannot be greater than max (%d)", min, max)
+	}
+
+	rangeSize := max - min + 1
+	// Calculate the largest multiple of rangeSize that fits in MaxInt64
+	limit := math.MaxInt64 - (math.MaxInt64 % rangeSize)
+
+	for {
+		n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
```
Comment (shahzadhaider1) [inline rand/rand.go:37]: why didn't we simply use `rand.Int(rand.Reader, big.NewInt(rangeSize))`?

Comment (kashifkhan0771) [reply rand/rand.go:37]: That's for generating number in a specific range that's why

File: rand/rand.go (hunk 2 above)
Comment (guisithos) [reply rand/rand.go:27]: Yeah that makes sense, thanks!

**Comment by guisithos** [comment]
I followed [coderabbitai](https://github.com/apps/coderabbitai) advices, did some changes and updated my files.

File: rand/rand_test.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+			}
+
+			if err == nil && !contains(tt.slice, got) {
+				t.Errorf("Pick() returned value %v not found in slice", got)
+			}
+		})
+	}
+}
+
+func TestShuffle(t *testing.T) {
```
Comment (kashifkhan0771) [inline rand/rand_test.go:172]: This test case is failing. Please take a look. Rest looks ok to me.

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]

**Review by kashifkhan0771 — APPROVED** [review]
LGTM! Thanks for the contribution @guisithos

**Pull request merged**